### PR TITLE
[Feature] Add a HINT for the per edge type sampler of heterogeneous DistGraph that highlighting the etypes are sorted already.

### PR DESCRIPTION
**Pull request by classicsong**
## Description
The per edge type sampling for heterogeneous DistGraph assume the etypes are sorted by default.
A test on a sensatic dataset that has 19717 nodes, 4 edge types and 40354604 edges shows that the overall sampling speed can be improved by 4~5%.

#3193 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/distributed/graph_services.py [hunk 1]
```diff
@@ -81,7 +81,8 @@ def _sample_etype_neighbors(local_g, partition_book, seed_nodes, etype_field,
     local_ids = F.astype(local_ids, local_g.idtype)
     # local_ids = self.seed_nodes
     sampled_graph = local_sample_etype_neighbors(
-        local_g, local_ids, etype_field, fan_out, edge_dir, prob, replace, _dist_training=True)
+        local_g, local_ids, etype_field, fan_out, edge_dir, prob, replace,
+        etype_sorted=True, _dist_training=True)
```
Comment (zheng-da) [inline python/dgl/distributed/graph_services.py:85]: where is this variable defined?

**Comment by zheng-da** [comment]
how does this PR pass the CI test? does the test not cover the function or the CI ignores the error?

File: python/dgl/distributed/graph_services.py (hunk 1 above)
Comment (classicsong) [reply python/dgl/distributed/graph_services.py:85]: It is defined in dgl.sampling.neighbor.py L16

**Comment by classicsong** [comment]
> how does this PR pass the CI test? does the test not cover the function or the CI ignores the error?

I have added more tests in C++.
For python test. The DistGraph always has sorted edges.

File: python/dgl/distributed/graph_services.py (hunk 1 above)
Comment (zheng-da) [reply python/dgl/distributed/graph_services.py:85]: ah, `etype_sorted` is the input argument name. i thought it's a variable in this function.

Comment (zheng-da) [reply python/dgl/distributed/graph_services.py:85]: is it guaranteed that the graph is sorted by etype in DistGraph?

Comment (zheng-da) [reply python/dgl/distributed/graph_services.py:86]: please write some comment that `DistGraph`'s edges are sorted.

**Comment by zheng-da** [comment]
please also add unit tests to verify that the edges of a heterogeneous graph after graph partitioning have been sorted based on the edge type.

**Comment by zheng-da** [comment]
otherwise, the PR looks good to me.

**Review by zheng-da — APPROVED** [review]

**Pull request merged**